test/Riok.Mapperly.Tests/Mapping/UserMethodTest.cs

```diff
@@ -44,14 +44,37 @@ public Task InstanceMapperShouldEmitDiagnosticForStaticMethods()
 [Mapper]
 public partial class MyMapper
 {
-    public partial static object StaticToObject(string s);
+    public static object StaticToObject(string s);
 
     public partial object InstanceToObject(string s);
 }
 ";
         return TestHelper.VerifyGenerator(source);
     }
 
+    [Fact]
+    public Task InstanceMapperShouldSupportUserDefinedStaticMethods()
+    {
+        var source =
+            @"
+using System;
+using System.Collections.Generic;
+using Riok.Mapperly.Abstractions;
+
+[Mapper]
+public partial class MyMapper
+{
+    public static int StaticMapper(int s);
```

Comment:
I'd try to keep the test cases valid c# when easily possible. Probably add a method expression here (eg. `=> s + 10`).